sphinx/util/logging.py

```diff
@@ -375,12 +375,8 @@ def is_suppressed_warning(type: str, subtype: str, suppress_warnings: List[str])
         else:
             target, subtarget = warning_type, None
 
-        if target == type:
-            if ((subtype is None and subtarget is None) or
-                    subtarget is None or
-                    subtarget == subtype or
-                    subtarget == '*'):
-                return True
+        if target == type and subtarget in [None, subtype, "*"]:
+            return True
```

Comment:
was scratching my head over this for a little while. wouldn't mind an extra pair of eyes or two to check my logic here!

Reply:
Did you drop the check for `subtype is None` on purpose?

This doesn't look right. Or am I missing something?

Reply:
> Did you drop the check for `subtype is None` on purpose?
> 
> This doesn't look right. Or am I missing something?

I did! here's the original code-

```python
        if target == type:
            if ((subtype is None and subtarget is None) or
                    subtarget is None or
                    subtarget == subtype or
                    subtarget == '*'):
                return True
```

the case you're referring to is `(subtype is None and subtarget is None)`. There's another `or` case which is `subtarget == subtype`. The second case includes the case that they're both `None`, hence `(subtype is None and subtarget is None)` is redundant and can be removed.

Reply:
Yeah, now I'm seeing it, thanks for the clarification!